### DISCUSSION TITLE
Cannot anymore publish to npm registry using yarn

### DISCUSSION THREAD

**Discussion by Jym77** [community/community#33025] [Actions] [answered]
I'm normally using `yarn npm publish` to publish packages to the Github npm registry. This used to work fine until August ([Last succesful run of the action](https://github.com/Siteimprove/alfa/actions/runs/2739869292)). It started breaking in September ([first failing run of the action](https://github.com/Siteimprove/alfa/actions/runs/2992904724)).

All I get is a 400 with error message "unmarshalling packument failed: json: cannot unmarshal string into Go struct field pcopy.versions of type npm.Bugs". I assume something changed on Github side and the accepted data is not exactly the same 🤔 

`npm publish`, on the other hand, works ([successful run of the new action](https://github.com/Siteimprove/alfa/actions/runs/3045516646/jobs/4907171005)). Sadly, it does not understand the `workspace` protocol in the manifest and thus produces packages that cannot be installed 😢 So it's not really a working solution for me.

I tried (locally) to first pack the package and then publish the tarball. Both `yarn pack` + `npm publish` and `npm pack` + `npm publish` fail with the same error as previously ("unmarshalling packument failed: json: cannot unmarshal string into Go struct field pcopy.versions of type npm.Bugs"). 
I'm actually fairly surprised that `npm pack` + `npm publish` fails the same way, it seems to indicate that npm is packing differently during `npm publish` alone. That is a bit unrelated, in the end Github package does not accept the result.

Given that I get a 400 rather than a 403, this is not an authentication problem, and the message doesn't look like a permission problem. So I assume my token has the correct permission (it has all `package` and `repo` permission, iirc).

The error message seems fairly different from other recent problems with publishing such as #30949 or #31022 (plus both of these seem to be resolved).

**Comment by tinaheidinger** [comment]
Hi @Jym77, we don't see yarn generally failing with the GitHub npm registry, so must be an issue specific to your package. Could you please share the package.json you're trying to publish?

**Reply by Jym77** [reply]
[this one](https://github.com/Siteimprove/alfa/blob/main/packages/alfa-highlight/package.json), for example:
```json
{
  "$schema": "http://json.schemastore.org/package",
  "name": "@siteimprove/alfa-highlight",
  "homepage": "https://alfa.siteimprove.com",
  "version": "0.44.5",
  "license": "MIT",
  "description": "Functionality for working with terminal string styling",
  "repository": {
    "type": "git",
    "url": "https://github.com/siteimprove/alfa.git",
    "directory": "packages/alfa-highlight"
  },
  "bugs": "https://github.com/siteimprove/alfa/issues",
  "main": "src/index.js",
  "types": "src/index.d.ts",
  "files": [
    "src/**/*.js",
    "src/**/*.d.ts"
  ],
  "dependencies": {
    "chalk": "^4.1.2"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://npm.pkg.github.com/"
  }
}
```
(this is a package at the bottom of my monorepo dep tree, so should be a simple one).

**Reply by Jym77** [reply]
And here is [the tarball I get with `yarn pack`](https://drive.google.com/file/d/1ZZFTS0YIeziD1nQBvvfxeojbT7WP9-mo/view?usp=sharing).

**Comment by nishthaGupta** [comment, marked as answer]
Hello from Github Package Registry Team,

The issue was due to `bugs` from package.json getting packed as a string using yarn which our registry didn't support. We have patched a fix for the same.
Please re-try and let us know if still any errors are faced.

**Comment by Jym77** [comment]
Hi,
things are working fine now 🎉 [successful action run](https://github.com/Siteimprove/alfa/actions/runs/3065965122/jobs/4950618308)

Thanks for the help!
